contracts/staking/GnosisTargetDispenserL2.sol

```diff
@@ -53,12 +55,25 @@ contract GnosisTargetDispenserL2 is DefaultTargetDispenserL2 {
     }
 
     /// @inheritdoc DefaultTargetDispenserL2
-    function _sendMessage(uint256 amount, bytes memory) internal override {
+    function _sendMessage(uint256 amount, bytes memory bridgePayload) internal override {
+        // Check for the bridge payload length
+        if (bridgePayload.length != BRIDGE_PAYLOAD_LENGTH) {
+            revert IncorrectDataLength(BRIDGE_PAYLOAD_LENGTH, bridgePayload.length);
+        }
+
+        // Get the gas limit from the bridge payload
+        uint256 gasLimitMessage = abi.decode(bridgePayload, (uint256));
+
+        // Check the gas limit value
+        if (gasLimitMessage < GAS_LIMIT) {
+            gasLimitMessage = GAS_LIMIT;
+        }
```

Comment:
@DavidMinarsch @mariapiamo re: L2 to L1 message passing of a single amount variable. The bridged payload is checked for the gas limit, which is always bound by the number sufficient for the message delivery from L2 to L1. The value passed can always be bigger than that limit.